libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -944,13 +941,7 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		vector<string> args;
 		if (functionType->bound())
-		{
-			solAssert(memberAccess && functionDef, "");
-			solAssert(functionDef->parameters().size() == arguments.size() + 1, "");
-			args += convert(memberAccess->expression(), *functionDef->parameters()[0]->type()).stackSlots();
-		}
-		else
-			solAssert(!functionDef || functionDef->parameters().size() == arguments.size(), "");
+			args += IRVariable(_functionCall.expression()).part("self").stackSlots();
```

Comment:
I think these assertions are all still true.

For example if `functionType->bound() && functionDef` then we have a pointer to a bound function which should be impossible.

Reply:
It is, when you have `(x.f)()` - as seen in one of the tests.

Reply:
Note that argument and parameter size comparison is enforced by the yul analysis, so it's not too much less strict.

Reply:
Oh. That's bad. You're right about `solAssert(memberAccess && functionDef, "")`. I remember that I spent a lot of time trying to come up with something that would fall into this branch when I was adding support for internal bound functions and I didn't find anything. Now I see that `(L.foo)()` would have done it. I need to see if that's handled properly now.

I think that the two assertions about `functionDef->parameters().size()` are still valid though?